src/components/Tabs/index.js

```diff
@@ -0,0 +1,223 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import styled from "styled-components";
+
+import { TabsProvider, TabsConsumer } from "./TabsProvider";
+import { Text } from "../Text";
+import { spacing, zIndex, colors, themes } from "../../theme";
+import { largeAndUp } from "../../theme/mediaQueries";
+
+const padding = spacing.moderate;
+
+const Container = styled.div.attrs({
+  className: props => (props.areTabsOverflow ? "overflowed" : null)
+})`
+  position: relative;
+  z-index: ${zIndex.reset};
+  width: 100%;
+  overflow: visible;
+
+  &:after {
+    content: "";
+    display: block;
+    position: absolute;
+    border-bottom: 1px solid ${themes.tm.gray04};
+    left: 0;
+    right: 0;
+    bottom: 0;
+    z-index: ${zIndex.unset};
+  }
+
+  &.overflowed:after {
+    width: 100vw;
+    left: calc(50% - 50vw);
+  }
+`;
+
+const Content = styled.div.attrs({
+  className: props => (props.areTabsOverflow ? "overflowed" : null)
+})`
+  display: flex;
+  flex: 1;
+  width: 100%;
+  overflow-x: auto;
+  scroll-behavior: smooth;
+
+  &.overflowed {
+    margin-left: calc(50% - 50vw);
+    padding-left: calc(50vw - 50%);
+    width: 100vw;
+  }
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+`;
+
+const Item = styled.div`
+  ${largeAndUp`
+    padding-right: ${spacing.comfy};
+  `};
+  display: block;
+  padding-right: ${spacing.normal};
+  white-space: nowrap;
+
+  &:last-of-type {
+    padding-right: 0;
+  }
+`;
+
+const checkIfOverflows = ({ offsetWidth = 0, scrollWidth = 0 } = {}) =>
+  offsetWidth < scrollWidth;
+
+const TabItemButton = styled.div.attrs({
+  className: props => (props.isActive ? "isActive" : ""),
```

Comment:
Can you please use BEM notation for all classnames? E.g. `.tab__button--active`

Reply:
Done. Thanks.

Reply:
Thank you!